### PR TITLE
Beta Part 3: Part of Mega Dynamic Load/Unload support

### PR DESCRIPTION
**Pull request by MSNev**
- Add Core SDK Unload support

### CODE REVIEW

File: shared/AppInsightsCore/src/JavaScriptSDK/BaseTelemetryPlugin.ts [hunk 1]
```diff
@@ -158,6 +161,8 @@ export abstract class BaseTelemetryPlugin implements ITelemetryPlugin {
                     }
                 }
             };
+
+            proxyFunctionAs(_self, "_addUnloadCb", () => _unloadHandlerContainer, "add");
```
Comment (MSNev) [inline shared/AppInsightsCore/src/JavaScriptSDK/BaseTelemetryPlugin.ts:165]: This exposes effectively the same function as
```typescript
_self._addUnloadCb = (<function arguments>) => {
    return _unloadHandlerContainer.add(<function arguments>);
};
```

The reason it's using a function to return the _unloadHandlerContainer is to support the value of this being changed without having to redefine the prototype function

File: shared/AppInsightsCore/src/JavaScriptSDK.Enums/TelemetryUnloadReason.ts [hunk 2]
```diff
@@ -13,15 +13,15 @@
...
      * This instance of the plugin is being removed and replaced
      */
-    //PluginReplace = 2,
+    PluginReplace = 2,
 
     /**
      * The entire SDK is being unloaded
      */
-    //SdkUnload = 50
+    SdkUnload = 50
```
Comment (hectorhdzg) [inline shared/AppInsightsCore/src/JavaScriptSDK.Enums/TelemetryUnloadReason.ts:26]: Big jump from 2 to 50, any specific reason for this one?

Comment (MSNev) [reply shared/AppInsightsCore/src/JavaScriptSDK.Enums/TelemetryUnloadReason.ts:26]: Yeh, I figured SDK unload should be separated from the other "plugin" related reasons

**Review by hectorhdzg — APPROVED** [review]
LGTM

**Pull request merged**